src/codegen_call_native.cpp

```
@@ -199,3 +260,217 @@ llvm::Value *CodeGen::emitTableDrivenNativeCall(
+        int depth = 0;
+        size_t commaPos = std::string::npos;
+        for (size_t i = 7; i < returnType.size(); ++i) {
+            if (returnType[i] == '<') ++depth;
+            else if (returnType[i] == '>') --depth;
+            else if (returnType[i] == ',' && depth == 0) { commaPos = i; break; }
+        }
+        if (commaPos == std::string::npos) return {RW::Direct, ""};
+        std::string okType = returnType.substr(7, commaPos - 7);
+        while (!okType.empty() && okType.back() == ' ') okType.pop_back();
+
+        if (okType == "Unit")  return {RW::ResultStatus, ""};
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)